doc/html/pcre2api.html

```diff
@@ -1807,85 +1916,48 @@ <h1>pcre2api man page</h1>
 <pre>
   PCRE2_NO_AUTO_POSSESS
 </pre>
-If this option is set, it disables "auto-possessification", which is an
-optimization that, for example, turns a+b into a++b in order to avoid
+If this (deprecated) option is set, it disables "auto-possessification", which
+is an optimization that, for example, turns a+b into a++b in order to avoid
 backtracks into a+ that can never be successful. However, if callouts are in
 use, auto-possessification means that some callouts are never taken. You can
 set this option if you want the matching functions to do a full unoptimized
 search and run all the callouts, but it is mainly provided for testing
-purposes.
+purposes. (It is recommended to use <b>pcre2_set_optimize</b> instead.)
```

Comment:
should mention the directive to use as well, and probably also mention that if both are given this one takes precedence.

the message should also be in its own paragraph instead

Reply:
OK, sounds good.

Reply:
Don't know why I can't add a reply to the above comment about `PCRE2_ERROR_BADOPTION`. Anyways, what I would like to say is that, while the other "set" functions return `PCRE2_ERROR_BADDATA` on invalid input, I feel that the symbolic name `BADDATA` does not describe the actual error well in this case. For other "set" functions it does make sense; for example, if the value passed for "maximum heap size" or "maximum variable lookbehind length" is invalid, we could (in natural human language) describe that as "bad data". In this case, my preferred name for the error code would be something like `PCRE2_UNKNOWN_DIRECTIVE`. Do you think it would be a good idea to add a new error code for this reason?

Reply:
The suggestion re: `PCRE2_ERROR_NULL` is fine and I have implemented it.